source/codegen/templates/proto_helpers.mako

```diff
@@ -51,45 +19,60 @@ enum ${service_class_prefix}Attributes {
   ${attribute_value_prefix}_${attribute_name} = ${attribute};
 % endfor
 }
+</%def>
 
+## Define enums in the proto for each metadata enum referenced in a proto message.
+<%def name="define_enums(used_enums)">\
+<%
+  enums = data["enums"]
+%>\
 % for enum_name in (e for e in enums if e in used_enums):
 <%
   enum_value_prefix = common_helpers.pascal_to_snake(enum_name).upper()
   enum = enums[enum_name]
-  allow_alias = proto_helpers.determine_allow_alias(enum)
+  allow_alias = proto_helpers.should_allow_alias(enum)
+  nonint_index = 1
 %>\
 enum ${enum_name} {
 % if allow_alias == True:
   option allow_alias = true;
 % endif
   ${enum_value_prefix}_UNSPECIFIED = 0;
-<%
-nonint_index = 1
-%>\
 % for value in enum["values"]:
-% if enum.get("generate-mappings", false):
+% if enum.get("generate-mappings", False):
   ${enum_value_prefix}_${value["name"]} = ${nonint_index};
 <%
-  nonint_index = nonint_index+1
+  nonint_index = nonint_index + 1
 %>\
 % else:
   ${enum_value_prefix}_${value["name"]} = ${value["value"]};
-%endif
+% endif
```

Comment:
Since mako allows for spaces between the `%` and the directive, it could sometimes help to use that spacing to indicate indentation levels. I previously did it somewhere in service.cpp where there were lots of nested if/else ladders.

Reply:
I did another pass through all the mako templates and helpers and standardized how the spacing is done. Now it's one space at first indent level and two additional spaces per level after that. This seems to read better and is consistent now at least.